central/networkpolicies/graph/graph_builder.go

```diff
@@ -112,11 +113,16 @@ func (b *graphBuilder) evaluatePeers(currentNS *storage.NamespaceMetadata, peers
 
 func (b *graphBuilder) evaluatePeer(currentNS *storage.NamespaceMetadata, peer *storage.NetworkPolicyPeer) []*node {
 	if peer.GetIpBlock() != nil {
-		// TODO(ROX-5370): We assume all CIDR blocks always match all deployments and overlapping external CIDRs.
-		//  This is probably wrong, but we don't really have a good way of determining otherwise. Except for maybe
-		//  look at Pod IPs. Which we actually could.
-		allNodes := make([]*node, 0, len(b.allDeployments)+1)
-		allNodes = append(allNodes, b.allDeployments...)
+		var allNodes []*node
+		// If the IP is in the private range, we add edges to other deployments in the cluster.
+		// This is still not the perfect approach, but it solves user issues where edges were being
+		// shown on *every* deployment with a Network Policy that uses CIDR block matchers. By
+		// limiting to private ranges only, the set of edges in the graph is likely more accurate, but
+		// still not the reality. There is an epic created to tackle this further and come up with a
+		// more elaborate logic for this: ROX-12120
+		if netutil.IsCIDRBlockInPrivateSubnet(peer.GetIpBlock().GetCidr()) {
```

Comment:
I forget what the outcome of our discussion was, but I would still say that overlap is the property that would need to be checked, not inclusion.

The way the code is currently written, `0.0.0.0/0` would _not_ result in edges to deployments being created.

Reply:
Changed to overlap. Thanks!